iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -718,4 +401,651 @@ TEST_F(ServiceDiscoveryPubSub_test, ServiceDiscoveryNotifiedbyListenerFindsSingl
+
+    this->testFindService({"Ferdinand"}, {"Schnüffel"}, {"Spitz"});
+}
+
+// Limit test of both kinds (MessagingPattern) of service
+//
+// 1) fill the discovery to the max with PUB/SUB and REQ/RES services
+// 2) vary services instances and events create a general scenario that can occur in practice:
+//    e.g. same service with many instances, same instance with many events etc.
+// As such, the tests check whether service discovery works at system limits,
+// i.e. findService works for a full service registry in a general case scenario
+TYPED_TEST(ServiceDiscoveryFindService_test, FindInMaximumMixedServices)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e0b292c2-49ce-47b4-b38e-456732240c41");
+    // maximum number of producers to generate
+    auto constexpr MAX = TestFixture::maxProducers(TestFixture::Variation::PATTERN);
+    // first threshold to change generation strategy at (about 1/3 of max)
+    auto constexpr N1 = MAX / 3;
+    // second threshold to change generation strategy at (about 2/3 of max)
+    auto constexpr N2 = 2 * N1;
+
+    // phase 1 : generate random services (no Umlaut)
+    auto s1 = randomService();
+    this->add(s1);
+
+    uint32_t created = 1;
+
+    for (; created < N1; ++created)
+    {
+        this->add(randomService());
+    }
+
+    // create a unique service presented by Umlaut (hence unique)
+    ServiceDescription s2{"Ferdinand", "Spitz", "Schnüffler"};
+    this->add(s2);
+    ++created;
+
+    // phase 2 : fix service, generate random instances (no Umlaut)
+    for (; created < N2; ++created)
+    {
+        this->add(randomService("Ferdinand"));
+    }
+
+    for (; created < MAX - 1; ++created)
+    {
+        this->add(randomService("Ferdinand", "Spitz"));
+    }
+
+    // phase 3 : fix service and instance, generate random events (no Umlaut)
+    auto s3 = randomService("Ferdinand", "Spitz");
+    this->add(s3);
+    ++created;
+
+    EXPECT_EQ(created, MAX);
+
+    // create some services of the other kind
+    created = 0;
+
+    auto constexpr OTHER_MAX = TestFixture::maxProducers(TestFixture::otherPattern());
+
+    // same phases, but now for the other service type
+    // note that thresholds N1, N2 are chosen in a way that we can reuse them
+    for (; created < N1; ++created)
+    {
+        this->addOther(randomService());
+    }
+
+    for (; created < N2; ++created)
+    {
+        this->addOther(randomService("Spitz"));
+    }
+
+    for (; created < OTHER_MAX; ++created)
+    {
+        this->addOther(randomService("Ferdinand", "Spitz"));
+    }
+
+    EXPECT_EQ(created, OTHER_MAX);
+
+    // now we have the maximum of services of both kinds
+    // with semi-random services
+
+    // search for specific services we inserted at various times (includes wildcard searches etc.):
+    // find first offered service, last offered service and some service offered inbetween
+    this->testFindService(s1);
+    this->testFindService(s2);
+    this->testFindService(s3);
```

Comment:
OK, I can add a comment. But the idea is that they set up a discovery (by adding services). Then call findService with all different combinations of wildcards or values (controlled by the parameter of the typed test) and compare the result with the reference implementation (which simply tracks the added services and hence knows the result).

For example `this->testFindService(s1)` in its simplest form check whether the sut and the reference implementation agree when we try to find service s1. It gets more complicated since the test is also executed with `s1` where say the instance is replaced by a wildcard. There are 8 combinations, and technically all of them need to be checked.

This is all to be able to write it very briefly as there are a lot of scenarios to be tested and I for one found the naming and structure of the previous tests inconsistent and there were a lot of cases missing.